leaf-ui/rappid-extensions/BackboneModels.js

```diff
@@ -1 +1,455 @@
-/** This file contains backbone model representations of the original model objects - WIP */
\ No newline at end of file
+/** This file contains backbone model representations of the original model objects - WIP */
+
+// FuncSegment Backbone Model from FuncSegment class in modelObjects.js
+var FunctionSegmentBBM = Backbone.Model.extend({
+    idAttribute: "uid",
+
+    // Initialize function - save all of the input parameters into one 'option' input
+    // And then use 'option' when intitializing the function 
+    initialize: function (options) {
+        this.type = options.type;
+        this.refEvaluationValue = options.refEvaluationValue;
+        this.startTP = options.startTP;
+        this.stopTP = options.stopTP;
+    },
+});
+
+// RepFuncSegment Backbone Model from RepFuncSegment class in modelObjects.js
+var RepFuncSegmentBBM = Backbone.Model.extend({
+    idAttribute: "uid",
+
+    initialize: function(options) { 
+        // this functionSegList is an array of all of the FunctionSegmentBBMs in the RepFuncSegmentBBM
+        this.functionSegList = options.functionSegList; 
+        this.repNum = options.repNum; 
+        this.absRepTime = options.absRepTime; 
+    }
+
+});   
+
+
+
+// Backbone Model of Evolving Function Class on modelObjects.js
+var EvolvingFunctionBBM = Backbone.Model.extend({
+    idAttribute: "uid",
+
+    initialize: function(options) { 
+        //this.intentionID = options.intentionID; 
+        this.nodeID = options.nodeID;  
+        // Not Sure if calling the other models inside this one is done correctly
+        var FuncSegment = new FuncSegmentModel();
+        var FuncSegmentOptions = ; // Have to eventually add all of the FuncSegment parameters here
+        FuncSegment.initialize(FuncSegmentOptions);
+        var RepFuncSegment = new RepFuncSegmentModel();
+        var RepFuncSegmentOptions = ; // Have to eventually add all of the RepFuncSegment parameters here
+        RepFuncSegment.initialize(RepFuncSegmentOptions);
+    }, 
+
+    defaults: { 
+        stringDynVis: 'NT',  
+        // There can only be one RepFuncSegment
+        // If there is a RepFuncSegment, input a RepFuncSegment model into this parameter
+        RepFuncSegment: null,
+        // Array functionSegList contains all of the FuncSegment models (and only FuncSegment models)
+        functionSegList: [], 

```

Comment:
We may potentially want this to be a collection and not an array but I think it can stay an array for now